### PR TITLE
tests: Add test to check if a block device is passed as volume.

### LINKED ISSUE #191: Add test to check if a block device passed as a volume is passed to VM using a block driver instead of 9p.
Test needs to be added after https://github.com/kata-containers/runtime/pull/138 lands.

### PR DESCRIPTION
**Pull request by GabyCT**
This will check if a block device passed as a volume is passed to VM using a block driver as well as it adds a tests that checks if a text file under /dev is passed to a container and check that contents are the same.

Fixes #191

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @GabyCT - please can you add more detail to the commit explaining why this (frankly seemingly bizarre) test is required? :smile:

**Comment by grahamwhaley** [comment]
:-) I did wonder if passing a dummy dev made with a `mknod` was more appropriate - so, yes, some more details would help clarify that.

File: integration/docker/volume_test.go [hunk 1]
```diff
@@ -90,4 +90,20 @@ var _ = Describe("docker volume", func() {
 			Expect(stdout).NotTo(ContainSubstring(testFile))
 		})
 	})
+
+	Context("creating a text file under /dev", func() {
+		It("should be passed the content to the container", func() {
+			fileName := "/dev/foo"
```
Comment (devimc) [inline integration/docker/volume_test.go]: you should remove it once the test finish

File: integration/docker/volume_test.go [hunk 2]
```diff
@@ -90,4 +96,58 @@ var _ = Describe("docker volume", func() {
...
 	})
+
+	Context("creating a text file under /dev", func() {
+		It("should be passed the content to the container", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create files under /dev")
+			}
+			fileName := "/dev/foo"
+			textContent := "hello"
+			err := ioutil.WriteFile(fileName, []byte(textContent), 0644)
```
Comment (devimc) [inline integration/docker/volume_test.go]: `err` was declared in the `var` block

File: integration/docker/volume_test.go [hunk 3]
```diff
@@ -90,4 +96,58 @@ var _ = Describe("docker volume", func() {
...
+		})
+	})
+
+	Context("passing a block device", func() {
+		It("should be mounted", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create loop devices")
+			}
+
+			for i := 0; i < loopDevices; i++ {
```
Comment (devimc) [inline integration/docker/volume_test.go]: `loopDevices` is 1, you don't need a loop to create 1 loopdevice

File: integration/docker/volume_test.go [hunk 4]
```diff
@@ -90,4 +96,58 @@ var _ = Describe("docker volume", func() {
...
+
+			for _, lf := range loopFiles {
+				err = deleteLoopDevice(lf)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			for _, df := range diskFiles {
+				err = os.Remove(df)
+				Expect(err).ToNot(HaveOccurred())
+			}
```
Comment (devimc) [inline integration/docker/volume_test.go]: you don't need two arrays (`loopFiles` and `diskFiles`) for 1 loop device

File: integration/docker/volume_test.go [hunk 5]
```diff
@@ -23,6 +23,12 @@ var _ = Describe("docker volume", func() {
 		fileTest      string = "hello"
 		exitCode      int
 		stdout        string
+		loopDevices   int = 1
```
Comment (devimc) [inline integration/docker/volume_test.go]: you don't need this variable

File: integration/docker/volume_test.go (hunk 4 above)
Comment (devimc) [reply integration/docker/volume_test.go]: `err = deleteLoopDevice(loopFile)` and `err = os.Remove(diskFile)`  should be enough

File: integration/docker/volume_test.go [hunk 6]
```diff
@@ -90,4 +93,50 @@ var _ = Describe("docker volume", func() {
...
+	Context("passing a block device", func() {
+		It("should be mounted", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create loop devices")
+			}
+
+			diskFile, loopFile, err = createLoopDevice()
+			Expect(err).ToNot(HaveOccurred())
+
+			args = []string{"--name", id, "--cap-add=SYS_ADMIN", "-v", loopFile + ":" + loopFile, DebianImage, "mount", loopFile, "/mnt"}
```
Comment (devimc) [inline integration/docker/volume_test.go]: to have `mount` working you have to format the loop device before mounting, so you have to do something like this

```go
loopFileP1 = fmt.Sprintf("%sp1", loopFile)
mkfsCmd := tests.NewCommand("mkfs.ext4", loopFileP1)
if _, stderr, exitCode := mkfsCmd.Run(); exitCode != 0 {
	return "", "", fmt.Errorf("%s", stderr)
}
args = []string{"--name", id, "--cap-add=SYS_ADMIN", "-v", loopFileP1 + ":" + loopFileP1, DebianImage, "mount", loopFileP1, "/mnt"}
```

Comment (GabyCT) [reply integration/docker/volume_test.go]: @devimc thanks for the feedback

Comment (devimc) [reply integration/docker/volume_test.go]: updated example code

**Comment by GabyCT** [comment]
@jodh-intel thanks for the feedback, I made the changes

**Comment by chavafg** [comment]
lgtm

**Pull request merged**